services/net/reporting/ReportingManager.cs

```diff
@@ -366,44 +379,57 @@ private async Task GenerateReportAsync(ReportRequestModel request, API.Areas.Ser
-                var response = await SendEmailAsync(request, to, subject, body, $"{report.Name}-{report.Id}");
+            var failed = false; // I don't like this implementation because one group of emails could be successful, but the other fail.
+            var responseModel = new ReportEmailResponseModel();
 
-                // Update the report instance.
-                if (request.GenerateInstance && instanceModel != null)
+            if (linkOnlyFormatTo.Any() || !String.IsNullOrEmpty(request.To))
+            {
+                try
+                {
+                    // Send the email.
+                    var response = await SendEmailAsync(request, linkOnlyFormatTo, subject, linkOnlyFormatBody, $"{report.Name}-{report.Id}-linkOnlyFormat");
+                    responseModel.LinkOnlyFormatResponse = JsonDocument.Parse(JsonSerializer.Serialize(response, _serializationOptions));
+                }
+                catch (ChesException ex)
                 {
-                    model.Id = instanceModel.Id;
-                    model.Version = instanceModel.Version ?? 0;
-                    model.SentOn = DateTime.UtcNow;
-                    model.Status = ReportStatus.Accepted;
-                    model.Content.ForEach(ric => ric.InstanceId = model.Id);
-                    model.Response = JsonDocument.Parse(JsonSerializer.Serialize(response, _serializationOptions));
-                    await this.Api.UpdateReportInstanceAsync(model);
+                    responseModel.LinkOnlyFormatResponse = JsonDocument.Parse(JsonSerializer.Serialize(ex.Data["error"], _serializationOptions));
+                    failed = true;
                 }
             }
-            catch (ChesException ex)
+            if (!failed && fullTextFormatTo.Any() || !String.IsNullOrEmpty(request.To))
             {
-                // Update the report instance.
-                if (request.GenerateInstance && instanceModel != null)
+                try
+                {
+                    // Send the email.
+                    var response = await SendEmailAsync(request, fullTextFormatTo, subject, fullTextFormatBody, $"{report.Name}-{report.Id}");
+                    responseModel.FullTextFormatResponse = JsonDocument.Parse(JsonSerializer.Serialize(response, _serializationOptions));
+                }
+                catch (ChesException ex)
                 {
-                    model.Id = instanceModel.Id;
-                    model.Version = instanceModel.Version ?? 0;
-                    model.Status = ReportStatus.Failed;
-                    model.Content.ForEach(ric => ric.InstanceId = model.Id);
-                    model.Response = JsonDocument.Parse(JsonSerializer.Serialize(ex.Data["error"], _serializationOptions));
-                    await this.Api.UpdateReportInstanceAsync(model);
+                    responseModel.FullTextFormatResponse = JsonDocument.Parse(JsonSerializer.Serialize(ex.Data["error"], _serializationOptions));
```

Comment:
I would put a comment here to denote what's happening.  My guess is that you are storing something to do with the error in the report instance for later analysis

Reply:
Yes we have always been capturing the CHES response in our report and notification instances.  I had to modify this to support capturing two responses is all.